src/main/java/task/Deadline.java

```diff
@@ -0,0 +1,45 @@
+package task;
+import exception.LeginEmptyTaskException;
+import exception.LeginMissingParamsException;
+
+public class Deadline extends Task{
+    protected String by;
+
+    private static String getDeadlineDescription(String input) throws LeginMissingParamsException,
+            LeginEmptyTaskException {
+        validityCheck(input);
+        return input.substring(input.indexOf(" ") + 1, input.indexOf("/by") - 1);
+    }
+
+    private static void validityCheck(String input) throws LeginMissingParamsException,
```

Comment:
You would want to use a verb here instead.
```suggestion
    private static void checkValidity(String input) throws LeginMissingParamsException,
```